### PR TITLE
Add `typescript-react-boilerplate`

### PR DESCRIPTION
**Pull request by lieone**
Inspired by [Announcing TypeScript support in Electron](https://electron.atom.io/blog/2017/06/01/typescript). Created from regular TS web app example for React. Main and renderer both process in pure TypeScript. Link to repo [typescript-react-boilerplate](https://github.com/lieone/typescript-react-boilerplate).

**By submitting this pull request, I promise I have read the [contributing guidelines](https://github.com/sindresorhus/awesome-electron/blob/master/contributing.md) twice and ensured my submission follows it. I realize not doing so wastes the maintainers' time that they could have spent making the world better. 🖖**

### CODE REVIEW

**Comment by sindresorhus** [comment]
What is [this](https://github.com/lieone/typescript-react-boilerplate/blob/2f3e1c5dac8380d9f6514f6c8bb4cf96e74ea38d/package.json#L62)?

**Comment by sindresorhus** [comment]
> Start the description with a capital and end with a full stop/period.

https://github.com/sindresorhus/awesome-electron/blob/master/contributing.md

**Comment by lieone** [comment]
> What is this?
Sorry, npm [issue](https://github.com/npm/npm/issues/17141). Cleaned.

**Comment by sindresorhus** [comment]
I think the boilerplate needs some more documentation. Also ensure it's targeting the latest version of TypeScript, React, Webpack, Electron. Some might have been updated since you submitted here.

File: readme.md [hunk 1]
```diff
@@ -222,6 +222,7 @@ Made with Electron.
 - [electron-quick-start](https://github.com/electron/electron-quick-start) - Clone the repo to try a simple app.
 - [bozon](https://github.com/railsware/bozon) - Scaffold, run, test, and package your app.
 - [electron-vue](https://github.com/SimulatedGREG/electron-vue) - Easily build your app with Vue and common plugins.
+- [typescript-react-boilerplate](https://github.com/lieone/typescript-react-boilerplate) - Main and renderer both in TypeScript
```
Comment (sindresorhus) [inline readme.md]: "Main and renderer" is not clear enough. Needs to say:

> Shows TypeScript usage in both the main and renderer process.

Also note the dot at the end.

**Pull request closed without merging**